The ``7-base_geometry`` module
===============================

Using the ``BaseGeometry`` class
---------------------------------

This is a test file for the ``BaseGeometry`` class defined in the
``7-base_geometry`` module. It tests the ``area()`` method and the
``integer_validator(name, value)`` method.


First, import the ``BaseGeometry`` class from the module:
>>> BaseGeometry = __import__('7-base_geometry').BaseGeometry

Test the area() method:
    >>> bg = BaseGeometry()
    >>> bg.area()
    Traceback (most recent call last):
    	...
    Exception: area() is not implemented

Test the integer_validator(self, name, value) method:

With a valid integer (does nothing)
    >>> bg.integer_validator("age", 45)

When ``value`` is not an integer
    >>> bg.integer_validator("float", 3.14)
    Traceback (most recent call last):
            ...
    TypeError: float must be an integer

    >>> bg.integer_validator("name", "John")
    Traceback (most recent call last):
            ...
    TypeError: name must be an integer

    >>> bg.integer_validator("list", [1, 2, 3])
    Traceback (most recent call last):
            ...
    TypeError: list must be an integer

    >>> bg.integer_validator("set", {4, 5, 6})
    Traceback (most recent call last):
            ...
    TypeError: set must be an integer

    >>> bg.integer_validator("check", True)
    Traceback (most recent call last):
            ...
    TypeError: check must be an integer

    >>> bg.integer_validator("None", None)
    Traceback (most recent call last):
            ...
    TypeError: None must be an integer

When ``value`` is not greater than 0
    >>> bg.integer_validator("distance", -5)
    Traceback (most recent call last):
    	    ...
    ValueError: distance must be greater than 0

    >>> bg.integer_validator("age", 0)
    Traceback (most recent call last):
    	    ...
    ValueError: age must be greater than 0

Passing incorrect number of arguments
    >>> bg.integer_validator("name")
    Traceback (most recent call last):
    	    ...
    TypeError: BaseGeometry.integer_validator() missing 1 required positional argument: 'value'

    >>> bg.integer_validator()
    Traceback (most recent call last):
    	    ...
    TypeError: BaseGeometry.integer_validator() missing 2 required positional arguments: 'name' and 'value'